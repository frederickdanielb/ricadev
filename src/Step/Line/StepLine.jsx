import StepLineItem from './StepLineItem';
import PropTypes from 'prop-types';
import { mapCloneElement } from '../../utils/Children';
import './Step.scss';
import { StepsContainer } from './styled';
const StepLine = ({ children, current, onClick = () => {}, linkeable }) => {
	const items = mapCloneElement(children, (item, index) => {
		const itemProps = {
			index: index,
			isActive: index === current,
			isCompleted: index < current,
			onClick: () => (!linkeable ? onClick(index) : null),
			...item.props,
		};
		return itemProps;
	});
	return (
		<StepsContainer>
			<ul className="progress-tracker progress-tracker--text progress-tracker--center m-0 mt-4">
				{items}
			</ul>
		</StepsContainer>
	);
};
StepLine.Item = StepLineItem;
StepLine.propTypes = {
	/** StepLine.Item components stored as child of the StepLine component */
	children: PropTypes.arrayOf(
		PropTypes.shape({
			type: PropTypes.oneOf([StepLine.Item]),
		}),
	),
	/** Current step */
	current: PropTypes.number.isRequired,
	/** Function to be called when a step is clicked */
	onClick: PropTypes.func,
};
export default StepLine;
