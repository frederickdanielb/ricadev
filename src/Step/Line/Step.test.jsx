import { cleanup, render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import history from '../../history';
import StepLine from './StepLine';

describe('StepLine component', () => {
	let container = null;
	beforeEach(() => {
		container = render(
			<Router history={history}>
				<StepLine current={3}>
					<StepLine.Item title={'Seleccionar'} linkeable path={'/hotels/search'} />
					<StepLine.Item title={'Seleccionar2'} />
					<StepLine.Item title={'Seleccionar3'} />
					<StepLine.Item title={'Seleccionar4'} />
				</StepLine>
			</Router>,
		);
	});
	afterEach(() => {
		cleanup();
	});
	it('renders correctly', () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
});
