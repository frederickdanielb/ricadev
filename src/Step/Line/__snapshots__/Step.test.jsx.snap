// Vitest Snapshot v1

exports[`StepLine component > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="fullwidth_"
  >
    <div
      class="container_ separator_"
    >
      <ul
        class="progress-tracker progress-tracker--text progress-tracker--center m-0 mt-4"
      >
        <li
          class="progress-step is-complete"
        >
          <a
            href="/hotels/search"
          >
            <div
              class="progress-marker"
            />
            <div
              class="progress-text"
            >
              <h4
                class="progress-title"
              >
                Seleccionar
              </h4>
            </div>
          </a>
        </li>
        <li
          class="progress-step is-complete"
        >
          <div
            class="progress-marker"
          />
          <div
            class="progress-text"
          >
            <h4
              class="progress-title"
            >
              Seleccionar2
            </h4>
          </div>
        </li>
        <li
          class="progress-step is-complete"
        >
          <div
            class="progress-marker"
          />
          <div
            class="progress-text"
          >
            <h4
              class="progress-title"
            >
              Seleccionar3
            </h4>
          </div>
        </li>
        <li
          class="progress-step is-active"
        >
          <div
            class="progress-marker"
          />
          <div
            class="progress-text"
          >
            <h4
              class="progress-title"
            >
              Seleccionar4
            </h4>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
